### DISCUSSION TITLE
Understand the performance impact of using a unmatched pair of real device and simulator' config

### DISCUSSION THREAD

**Discussion by wangyuyue** [accel-sim/accel-sim-framework#161] [Q&A] [answered]
Hi developers,

thanks for providing this wonderful tool.

I have a question about using this tool:

Sometimes the real device on my hand does not have a tested simulator configuration in the repo, and I don't know how to write a reliable new configuration. Or I don't have the device I want to simulate, instead just a similar one.

In such a situation, I use `real device A` to generate the trace and then run the trace in the simulator using `device B`'s configuration. Does the performance result correspond to either `device A` or `device B`? Or worst, neither of them?

From previous issues, I know the simulator can generate results if the real device and simulator configuration have the same architecture, such as both Pascal. But I don't know what the result shows when they are different models of the same architecture.

I hope you'll be able to solve my confusion. Thanks a lot.

**Comment by JRPan** [comment, marked as answer]
I moved this to the discussion as this is not really a bug. 

So even though it's not recommended, you can mix use traces that are are generation.

The performance results correspond to the config. Not the device you collected traces on. 

Again, even though this is not recommended, you should be able to run traces with configs that are the same generation (Volta, Turing, Ampere, etc) without error. And results should be similar. 
An example could be running a 3090 config with traces collected on 3070. But running a 3090 config with traces collected in 2080 is a bad idea (and it probably won't even run). 

If you just want to create a config for some cards we don't have, check out the tuner under `util/tuner`. It generates a config by querying the GPU info and a few microbenchmarks. That's what we use. 

Thanks,
JR

**Comment by wangyuyue** [comment]
Thanks for your quick response.

I have another question: In **Accel-Sim SASS Frontend and Simulation Engine** section of README, there is a command

`/gpu-simulator/bin/release/accel-sim.out -trace ./hw_run/rodinia_2.0-ft/9.1/backprop-rodinia-2.0-ft/4096___data_result_4096_txt/traces/kernelslist.g -config ./gpu-simulator/gpgpu-sim/configs/tested-cfgs/SM7_QV100/gpgpusim.config -config ./gpu-simulator/configs/tested-cfgs/SM7_QV100/trace.config`

Why do we use two config files here? I find that they have different contents. Could you explain how to understand these two config files and what are they used for? Thanks very much.

**Reply by JRPan** [reply]
That should be a typo. Only 1 config is needed. We'll fix it.

**Reply by JRPan** [reply]
Actually, I lied. The configs are different. One is for Accel-sim and the other is for gpgpu-sim.
